website/docs/components.md

```diff
@@ -0,0 +1,146 @@
+
+## API
+
+### component
+
+The `component` method follows the same API as the `variant` method and allows you to apply default styles to any Kodiak UI components. The main difference is that Kodiak UI components will automatically check the theme for a style definition based on the component name.
+
+```ts
+component('key', { bg: 'primary', px: 4 })
+```
+
+| Argument | Type   | Description                                                                      |
+| -------- | ------ | -------------------------------------------------------------------------------- |
+| key      | string | The name of the component and how you will reference the styles in the base prop |
+| styles   | object | Theme scoped style object. The styles that will be applied to the component      |
+
+💡**Tip**: See the alternate way to define components [below](#defining-components-globally).
+
+### base prop
+
+All Kodiak UI components accept a `base` prop which will take a string or array of strings as a value. The component(s) passed into the `base` prop will apply the appropriate styling to the component. A Kodiak UI component will automatically check for a `component` defined in the theme. It will look to see if there a component in the theme with the same name as the component, only camelcased.
+
+```
+Button => button
+AccordionItem => accordionItem
+```
+
+```tsx
+<Button base="linkButton">Click me!</Button>
+
+// Automatically styled if a `button` component has been defined
+<Button>Click me!</Button>
+
+// Compose multiple components togther for more specific use-case versions of Kodiak's primitives
+<Button base={['button', 'link']}>Click me!</Button>
+```
+
+### useComponent
+
+The `useComponent` hook allows you to get the styles for a specified component outside of the normal parent -> child hierarchy. The hook accepts a component as the only argument and will return the styles from the theme that can be passed into the `sx` prop in any of Kodiak UI's components.
+
+| Argument  | Type                            | Description |
+| --------- | ------------------------------- | ----------- |
+| component | { key: string, styles: object } |             |
+
+```ts
+import { button } from './some-other-file'
+
+function App() {
+  const styles = useComponent(button)
+
+  return <Box sx={{ ...styles }}>Stuff</Box>
+}
+```
+
+### useComponents
+
+The `useComponents` hook will return all currently defined variants in the theme.
+
+🚨 **Note:** If you define a `component` in a React component that is not currently being rendered, it will not have been injected into the theme at this point and thus will not be returned by `useComponents`.
+
+```ts
+const components = useComponents()
+```
+
+## Defining components globally
+
+If you would prefer to define your components in one single place, you can pass them into the optional `components` object when intializing Kodiak UI.
+
+```ts
+const { theme } = createDesignSystem({
+  components: {
+    button: { // styles },
```

Comment:
`component` is the starting default style for named components.  Then the variant layers on top of the `component` right?
the order is:

```
base,
component,
variant,
sx
```

Reply:
Yep, exactly.